### PR TITLE
The pull request updates the navigation bar to enhance functionality, user experience, and design

### PR DESCRIPTION
**Pull request by Alimumtaz123**
Key changes include:
**Improved Layout**: Updated the structure to ensure better responsiveness across devices.

 Styling Enhancements:

-  Optimized color scheme for better visibility.

- Added hover effects for a more interactive experience.

Bug Fixes:

- Resolved alignment issues on smaller screens.

- Fixed inconsistencies in padding and margins.

### CODE REVIEW

File: home/templates/includes/navigation.html [hunk 1]
```diff
@@ -271,96 +247,96 @@
...
-    </div>
-    <div class="form-check form-switch dark-mode res-hide">
-      <input id="darkmode-toggle" class="form-check-input" type="checkbox" role="switch" id="flexSwitchCheckChecked">
-      <label class="form-check-label" for="flexSwitchCheckChecked">Dark Mode</label>
+        </span>
+      </div>
+      <!-- <div class="form-check form-switch dark-mode res-hide">
+        <input id="darkmode-toggle" class="form-check-input" type="checkbox" role="switch" id="flexSwitchCheckChecked">
+        <label class="form-check-label" for="flexSwitchCheckChecked">Dark Mode</label>
+      </div> -->
```
Comment (s-bonnor) [inline home/templates/includes/navigation.html:337]: So, the dark mode toggle is sort of important, could you fix that

**Review by s-bonnor — CHANGES_REQUESTED** [review]
Looks good other than the missing dark mode toggle

**Comment by Alimumtaz123** [comment]
Dark mode button added

**Comment by Alimumtaz123** [comment]
dark mode button added

**Comment by jbm1412** [comment]
Current Team Member's work so far:

Screenshot (1994)

**Review by s-bonnor — CHANGES_REQUESTED** [review]
Looks good except for 2 issues
1. the sign-in and sign-up dropdowns still show up when logged in
2. the sign-in and sign-up dropdowns dont respect dark mode and always have yellow text on a white background

also could you fix the merge conflicts

**Review by jbm1412 — CHANGES_REQUESTED** [review]
Screenshot (2032)
 
Tested -> Works. Looks Good. Please fix the merge conflicts @Alimumtaz123

**Comment by Alimumtaz123** [comment]
merge conflicts solved

**Comment by jbm1412** [comment]
Screenshot (2032)

Tested -> Works. Looks Good!

**Review by jbm1412 — APPROVED** [review]

**Review by s-bonnor — CHANGES_REQUESTED** [review]
Cyber Challenges Page is cutoff by the navigation bar
image

Search Icon is out of place on the about page
image

it doesnt seem to be possible to interact with the navigation bar at all in the mobile view (when the page is too narrow)

**Pull request merged**